modules/Bio/EnsEMBL/Compara/RunnableDB/OrthologQM/CalculateWGACoverage.pm

```diff
@@ -144,7 +144,11 @@ sub run {
 		if ( defined $exon_ranges ){
 			foreach my $aln_mlss ( keys %{ $this_aln_range } ){
 				foreach my $gdb_id ( sort {$a <=> $b} keys %{ $orth_ranges } ){
-					my $combined_coverage = $self->_combined_coverage( $orth_ranges->{$gdb_id}, $this_aln_range->{$aln_mlss}->{$gdb_id}, $exon_ranges->{$gdb_id} );
+					# Make sure that if this is a component gdb_id it refers back to the principal. The exon data is all on the components and not on the principal, but the ortholog info and aln_mlss is only on principal
+					my $gdb = $gdba->fetch_by_dbID($gdb_id);
+					my $principal_gdb = $gdb->principal_genome_db;
+					my $principal_gdb_id = $principal_gdb ? $principal_gdb->dbID : $gdb_id;
+					my $combined_coverage = $self->_combined_coverage( $orth_ranges->{$gdb_id}, $this_aln_range->{$aln_mlss}->{$principal_gdb_id}, $exon_ranges->{$gdb_id} );
```

Comment:
You said in the comment (line 147) that ortholog info is on the principal, so shouldn't this be
```suggestion
					my $combined_coverage = $self->_combined_coverage( $orth_ranges->{$principal_gdb_id}, $this_aln_range->{$aln_mlss}->{$principal_gdb_id}, $exon_ranges->{$gdb_id} );
```
?

Reply:
I need to alter the comment, I counted the ortholog information under protein trees